### PR TITLE
Update styling of date picker and calendar

### PR DESCRIPTION
**Pull request by origami-z**
#3530

### CODE REVIEW

**Comment by ivan-calderon** [comment]
All looking good @origami-z. Thanks! — [Design review link](https://www.figma.com/design/a67yyPmdOIJCM4X4d5Y3er/Calendar?node-id=4380-16&t=K8Meu1YmLljxTOpd-1).

File: packages/lab/src/calendar/internal/CalendarDay.css [hunk 1]
```diff
@@ -44,8 +44,21 @@
...
   z-index: var(--salt-zIndex-default);
+
+  border-top-left-radius: var(--salt-palette-corner-weak, 0);
+  border-bottom-left-radius: var(--salt-palette-corner-weak, 0);
+}
+
+.saltCalendarDay-selectedEnd {
+  border-top-right-radius: var(--salt-palette-corner-weak, 0);
+  border-bottom-right-radius: var(--salt-palette-corner-weak, 0);
 }
```
Comment (joshwooding) [inline packages/lab/src/calendar/internal/CalendarDay.css:62]: Is this separate due to specificity issues?

Comment (origami-z) [reply packages/lab/src/calendar/internal/CalendarDay.css:62]: ❔ do you mean start v.s. end? left vs right ?

Comment (joshwooding) [reply packages/lab/src/calendar/internal/CalendarDay.css:62]: sorry, I'll be less ambiguous :P `.saltCalendarDay-selected` already exists but this isn't with the previous styles

**Review by ivan-calderon — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**